components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -2662,7 +2672,7 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
                 throw new Error("Operation not implemented.");
             }
 
-            const result = await this.guardOrganizationOperationWithCentralizedPerms(teamId, fineGrainedOp);
+            const result = await this.guardOrganizationOperationWithCentralizedPerms(team.id, fineGrainedOp);
             if (result.err) {
                 throw result.err;
```

Comment:
Unrelated to this PR, but wanted to say that it's non-idiomatic (and therefore a bit confusing) to return error objects. Also looking at the implementation I would have expected that system configuration errors are thrown directly (i.e. no client or no feature flag) but that the actual check is a return value (no Error, though)

Reply:
This is really done for the case of the double-write and parallel read mode where not handling the exception as an extra case makes our lives a lot easier (since we don't actually rely on the system for checks yet). The intent is for this to normally throw on errors, once we default to perm checks with spicedb.

Feel free to refactor (and I started with that version) but the logic got even more complicated, as we needed to distinguish between system errors, feature not being enabled, the underlying client not existing (for system reasons) and finally not having permissions as seen by spicedb.